week05/my-todolist/src/StateTest.js

```diff
@@ -0,0 +1,11 @@
+import React, { useState } from "react";
+
+const StateTest = ({ tempState, setTempState }) => {
+  return (
+    <button id="testBtn" onClick={() => setTempState(tempState + 1)}>
+      테스트 버튼
+    </button>
+  );
+};
```

Comment:
P5: 현재는 tempState를 변경하는 방법으로 jsx 내부에서 익명함수를 사용하고 있지만, 이는 StateTest가 렌더링 될 때마다 리렌더링되므로 비효율적일 수 있습니다.
다음과 같이 handleTempState 함수를 생성하고, useCallback 훅을 사용하면 처음에만 handleTempState 함수가 리렌더링됩니다.


```suggestion
const StateTest = ({ setTempState }) => {
  const handleTempState = useCallback(() => {
    setTempState((prev) => prev + 1);
  }, []);

  return (
    <button id='testBtn' onClick={handleTempState}>
      테스트 버튼
    </button>
  );
};
```